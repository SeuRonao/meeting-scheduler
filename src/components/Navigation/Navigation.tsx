import React from "react";
import { Button, Container, Nav, Navbar } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import { useUserAuth } from "../../contexts/Auth/UserAuthContext";

export default function Navigation() {
  const { t } = useTranslation();
  return (
    <Navbar bg="light" variant="light">
      <Container>
        <Navbar.Brand href="/">{t("Navigation.brand")}</Navbar.Brand>
        <Nav className="me-auto">
          <Nav.Link href="/">{t("Navigation.links.home")}</Nav.Link>
          <Nav.Link href="/schedule">{t("Navigation.links.schedule")}</Nav.Link>
          <Nav.Link href="/clients">{t("Navigation.links.clients")}</Nav.Link>
        </Nav>
        <Profile />
      </Container>
    </Navbar>
  );
}

function Profile() {
  const { t } = useTranslation();
  const user = useUserAuth().user;
  if (user) {
    return <div>{user.displayName}</div>;
  } else {
    return (
      <Button variant="outline-success" href="/signIn">
        {t("Profile.use")}
      </Button>
    );
  }
}
