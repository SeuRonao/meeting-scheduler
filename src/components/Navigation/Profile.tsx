import React from "react";
import { signOut } from "firebase/auth";
import { Stack, Button, Alert } from "react-bootstrap";
import { useAuthState } from "react-firebase-hooks/auth";
import { useTranslation } from "react-i18next";
import { useNavigate } from "react-router-dom";
import { auth } from "../../utils/firebase/firebaseInit";
import LoadingSpinner from "../common/LoadingSpinner";

export default function Profile() {
  const { t } = useTranslation();
  const [user, loading, error] = useAuthState(auth);
  const navigate = useNavigate();

  async function handleLogOut() {
    try {
      await signOut(auth);
      navigate("/");
    } catch (error) {
      console.log(error);
    }
  }

  if (error) {
    return <Alert variant="danger">{error.message}</Alert>;
  }

  if (user) {
    return (
      <Stack direction="horizontal" gap={2}>
        <span>{user.displayName || user.email}</span>
        <Button variant="outline-secondary" size="sm" onClick={handleLogOut}>
          {t("Profile.sign-out")}
        </Button>
      </Stack>
    );
  }

  if (loading) {
    return <LoadingSpinner />;
  }

  return (
    <Button variant="outline-success" href="/sign-in">
      {t("Profile.sign-in")}
    </Button>
  );
}
