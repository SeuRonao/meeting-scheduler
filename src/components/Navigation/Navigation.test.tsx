import { MemoryRouter } from "react-router-dom";
import { create, act, ReactTestRenderer } from "react-test-renderer";
import Navigation from "./Navigation";

describe("Testing Navigation", () => {
  test("renders correctly", async () => {
    let component: ReactTestRenderer;
    await act(async () => {
      component = create(
        <MemoryRouter>
          <Navigation />
        </MemoryRouter>
      );
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
