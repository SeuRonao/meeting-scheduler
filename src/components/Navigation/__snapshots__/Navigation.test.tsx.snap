// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navigation renders correctly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-light bg-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      brand
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="me-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key={null}
          disabled={false}
          href="/"
          onClick={[Function]}
        >
          links.home
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key={null}
          disabled={false}
          href="/schedule"
          onClick={[Function]}
        >
          links.schedule
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key={null}
          disabled={false}
          href="/clients"
          onClick={[Function]}
        >
          links.clients
        </a>
      </div>
      <a
        className="btn btn-outline-success"
        href="/sign-in"
        onClick={[Function]}
      >
        sign-in
      </a>
    </div>
  </div>
</nav>
`;
